packages/components/src/Badge/Badge.component.js

```diff
@@ -16,6 +16,47 @@ const SIZES = {
 	small: 'small',
 };
 
+const DefaultBadge = ({ aslink, category, disabled, icon, id, label, onDelete, t }) => (
+	<React.Fragment>
+		{category && <BadgeLib.Category label={category} />}
+		{category && <BadgeLib.Separator />}
+		<BadgeLib.Label aslink={aslink} category={category} label={label}>
+			{icon && <BadgeLib.Icon name={icon} />}
+		</BadgeLib.Label>
+		{icon && onDelete && <BadgeLib.Separator iconSeparator />}
+		{onDelete && <BadgeLib.DeleteAction id={id} onClick={onDelete} disabled={disabled} t={t} />}
+	</React.Fragment>
+);
+
+DefaultBadge.propTypes = {
+	aslink: PropTypes.bool,
+	category: PropTypes.string,
+	disabled: PropTypes.bool,
+	icon: PropTypes.string,
+	id: PropTypes.string,
+	label: PropTypes.string,
+	onDelete: PropTypes.func,
+	t: PropTypes.func.isRequired,
+};
+
+// eslint-disable-next-line no-confusing-arrow
+const BadgeType = ({ disabled, onSelect, children, ...rest }) =>
+	onSelect ? (
+		<button {...rest} key="button" type="button" disabled={disabled} onClick={onSelect}>
+			{children}
+		</button>
+	) : (
+		<div {...rest} key="div">
+			{children}
+		</div>
+	);
```

Comment:
Could you write something like that in order to remove the `eslint-disable`, please ?

```javascript
const BadgeType = ({ disabled, onSelect, children, ...rest }) => {
	if (onSelect) {
		return (
			<button {...rest} key="button" type="button" disabled={disabled} onClick={onSelect}>
				{children}
			</button>
		);
	}

	return (
		<div {...rest} key="div">
			{children}
		</div>
	);
}
```